src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -67,6 +73,16 @@ public class JsonMessageFormatter : IJsonRpcMessageTextFormatter
         /// </summary>
         private readonly SequenceTextReader sequenceTextReader = new SequenceTextReader();
 
+        /// <summary>
+        /// Dictionary used to map the request id to their progress token so that the progress objects are cleaned after getting the final response.
+        /// </summary>
+        private readonly Dictionary<object, long> requestProgressMap = new Dictionary<object, long>();
```

Comment:
>object [](start = 36, length = 6)

Our outbound request IDs will always be `long`. The responses *might* convert that to a string. To to just make it very clear that you're not missing a dictionary key that should be equivalent, and to reduce memory pressure from boxing, let's store the key here as a `long` as well.